documentation/src/high-level/pauliz.md

```diff
@@ -2,21 +2,80 @@
 
-The `constant_circuit` will be used to prepare the state \\( |\psi> \\) by applying the Hadamard gate. The given Hamiltonian includes `X` and `Z` terms that cannot be measured at the same time, since they are measured using different bases. The `circuits` list includes one quantum circuit that does not apply any additional gate and one circuit that rotates the qubit basis into the X-basis so that the expectation value `<X>` is equivalent to the measurement of `<Z>` in the new basis. In this example, each measured Pauli product contains only one Pauli operator. For the post-processing of the measured results, the `PauliZProduct` measurement needs two more inputs provided by the object `PauliZProductInput`:
+ We define bit-registers `"ro_z"` and `"ro_x"` to store the measurements in the \\(Z\\)- and \\(X\\)-basis respectively. The `PragmaRepeatedMeasurement` operation is used to declare that we are making `1000` measurements of each circuit.
+
+```python
+# Z-basis measurement circuit with 1000 shots
+z_circuit = Circuit()
+z_circuit += ops.DefinitionBit("ro_z", 1, is_output=True)
+z_circuit += ops.PragmaRepeatedMeasurement("ro_z", 1000, None)
+
+# X-basis measurement circuit with 1000 shots
+x_circuit = Circuit()
+x_circuit += ops.DefinitionBit("ro_x", 1, is_output=True)
+# Changing to the X basis with a Hadamard gate
+x_circuit += ops.Hadamard(0)
+x_circuit += ops.PragmaRepeatedMeasurement("ro_x", 1000, None)
+ ```
+
+For the post-processing of the measured results, the `PauliZProduct` measurement needs two more inputs provided by the object `PauliZProductInput`:
 
 * The definition of the measured Pauli products after basis transformations (`add_pauliz_product()`),
 * The weights of the Pauli product expectation values in the final expectation values (`add_linear_exp_val()`).
 
-In general, one can measure the expectation values of the products of local Z operators, *e.g.* `<Z0>`, `<Z1>`, `<Z0*Z1>`, `<Z0*Z3>`, *etc.* The `PauliZProductInput` needs to define all of the products that are measured. In the given example, we will measure two products `<Z0>` after a rotation in the X basis (corresponding to `<X0>`) and `<Z0>` _without_ a rotation before the measurement.
-The `PauliZProductInput` also defines the weights of the products in the final result. In the example below, 0.1 is the coefficient for the first product and 0.2 for the second.
+The `PauliZProductInput` needs to define all of the products that are measured. In the given example, we will measure two products \\(\langle Z_0 \rangle\\) - one after a rotation in the \\(X\\)-basis (corresponding to \\(\langle X_0 \rangle\\)) and \\(\langle Z_0 \rangle\\) _without_ a rotation before the measurement.
```

Comment:
I understand what you mean here but I would maybe try to rephrase it to make it a bit clearer